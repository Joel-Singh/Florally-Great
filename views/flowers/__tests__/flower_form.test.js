const renderPugToDOM = require("../../renderPugToDOM.js");
const getValidRegionData = require("./../../../utils/dummyData/getValidData/getValidRegionData.js");

function renderRegionSelect(regionList) {
  const renderedForm = renderPugToDOM("./views/flowers/flower_form.pug", {
    regionList,
  });

  const regionListElement = renderedForm.querySelector(
    "[placeholder='select region']"
  );

  return regionListElement;
}

describe("testing generate region select", () => {
  test("Region select is empty with no regions", () => {
    expect(renderRegionSelect([])).toMatchSnapshot();
  });

  test("Region select has one region", () => {
    expect(renderRegionSelect([getValidRegionData()])).toMatchSnapshot();
  });

  test("Region select has multiple regions", () => {
    expect(
      renderRegionSelect([
        getValidRegionData(),
        getValidRegionData(),
        getValidRegionData(),
      ])
    ).toMatchSnapshot();
  });
});

describe("Prepopulated input values", () => {
  test("Display default values with no passed in preopulated values", () => {
    expect(getDefaultInputValueAttributesInFlowerForm({}))
      .toMatchInlineSnapshot(`
      [
        "",
        "",
        "",
        "",
      ]
    `);
  });

  test("Display prepopulated values", () => {
    const locals = {
      prepopulatedValues: {
        name: "prepopulated name",
        description: "preopopulated description",
        numberInStock: "32",
        price: "$3.21",
      },
    };

    expect(getDefaultInputValueAttributesInFlowerForm(locals))
      .toMatchInlineSnapshot(`
      [
        "prepopulated name",
        "preopopulated description",
        "32",
        "$3.21",
      ]
    `);
  });

  function getDefaultInputValueAttributesInFlowerForm(locals) {
    const renderedForm = renderPugToDOM(
      "./views/flowers/flower_form.pug",
      locals
    );
    let inputs = renderedForm.querySelectorAll("input");
    inputs = Array.from(inputs);

    const inputValues = inputs.map((element) => {
      return element.getAttribute("value");
    });

    return inputValues;
  }
});
