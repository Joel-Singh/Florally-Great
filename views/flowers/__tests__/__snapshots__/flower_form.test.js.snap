// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing generate region select Region select has multiple regions 1`] = `
<select
  id="region"
  name="regionID"
  placeholder="select region"
>
  <option
    value="000000000000000000000004"
  >
    Name3
  </option>
  <option
    value="000000000000000000000006"
  >
    Name5
  </option>
  <option
    value="000000000000000000000008"
  >
    Name7
  </option>
</select>
`;

exports[`testing generate region select Region select has one region 1`] = `
<select
  id="region"
  name="regionID"
  placeholder="select region"
>
  <option
    value="000000000000000000000002"
  >
    Name1
  </option>
</select>
`;

exports[`testing generate region select Region select is empty with no regions 1`] = `
<select
  id="region"
  name="regionID"
  placeholder="select region"
/>
`;
